install_script.sh.template

```diff
@@ -378,6 +378,212 @@ function ensure_config_file_exists() {
     fi
 }
 
+function report_installer_telemetry() {
+    local trace_id="$1"
+    local os="$2"
+    local distribution="$3"
+    local start_time="$4"
+    local exit_code="$5"
+    local install_stdout="$6"
+    local install_stderr="$7"
+
+    local time_now_seconds
+    local time_now
+    local time_since_start
+    local log_lines_stderr
+    local log_lines_stdout
+    local json_logs
+    local telemetry_trace
+    local telemetry_logs
+
+    echo -e "\033[34m\n* Reporting installer telemetry\n\033[0m"
+
+    if [ "$DD_INSTRUMENTATION_TELEMETRY_ENABLED" == "false" ] || \
+      [ "$site" == "ddog-gov.com" ] || \
+      [ -z "${apikey}" ] || \
+      [ -z "$telemetry_url" ]; then
+      return
+    fi
+
+    time_now_seconds=$(date +%s)
+    time_now=$(date +%s%N)
+    time_since_start=$((time_now - start_time))
+
+    telemetry_trace=$(cat <<-END
+    {
+        "api_version": "v2",
+        "request_type": "traces",
+        "tracer_time": ${time_now_seconds},
+        "runtime_id": "${trace_id}",
+        "seq_id": 1,
+        "origin": "linux-install-script",
+        "host": {
+            "hostname": "$(uname -n)",
+            "os": "${os}",
+            "distribution": "${distribution}",
+            "architecture": "$(uname -m)",
+            "kernel_version": "$(uname -v)",
+            "kernel_name": "$(uname -s)",
+            "kernel_release": "$(uname -r)"
+        },
+        "application": {
+            "service_name": "datadog-linux-install-script",
+            "service_version": "${install_script_version}",
+            "language_name": "UNKNOWN",
+            "language_version": "n/a",
+            "tracer_version": "n/a"
+        },
+        "payload": {
+            "traces": [[
+                {
+                    "service": "datadog-linux-install-script",
+                    "name": "install_installer",
+                    "resource": "install_installer",
+                    "trace_id": ${trace_id},
+                    "span_id": ${trace_id},
+                    "parent_id": 0,
+                    "start": ${start_time},
+                    "duration": ${time_since_start},
+                    "error": ${exit_code},
+                    "meta": {
+                        "language": "shell",
+                        "exit_code": ${exit_code},
+                        "error.message": "${install_stderr}"
+                    },
+                    "metrics": {
+                        "_top_level": 1
+                    }
+                }
+            ]]
+        }
+    }
+END
+)
+
+    printf -v log_lines_stderr "%b" "$install_stderr"
+    printf -v log_lines_stdout "%b" "$install_stdout"
+    json_logs="["
+    while IFS= read -r line; do
```

Comment:
to avoid surprises you can set back to previous IFS value
add a OLD_IFS=$IFS and set it back after

Reply:
I don't think it will override it, it's the usual `ENV_VAR command`, not an assignment